libs/@guardian/libs/src/consent-management-platform/Testing.md

```diff
@@ -0,0 +1,53 @@
+# Testing Banner changes
+
+This guide describes how to test changes to the CMP logic in Dotcom Rendering and Frontend, both manually and automatically using the monitoring tool.
+
+## Test locally
+
+1. In this repo, @guardian/csnx, run `make github-pages:dev`. This spins up a test page where you can interact with each banner and check the relevant consent state.
+
+## Test in CODE/PROD
+
+1. Create beta version by adding the `Canaries` label to your PR and wait for the beta version of @guardian/libs to be posted.
+2. Install the latest beta version in dotcom-rendering and frontend. Deploy branches of dotcom-rendering and frontend to CODE.
+
+### Automatic Tests using monitoring tool
+
+1. Setup the monitoring tool in https://github.com/guardian/consent-management-platform
+2. Use a vpn to test the different frameworks
+   - US for `usnat`
+   - GB for `tcfv2_corp` (Consent or Pay TCFV2)
+   - AUS for `aus`
+   - Ireland/ROW for `tcfv2`
+3. Execute `make synthetics-start-debug` in the @guardian/consent-management-platform and select the appropruate settings for the CODE environment and relevant framework
+
+### Manual checks to the website
+
+1. Use a vpn to test the different frameworks
+   - US for `usnat`
+   - GB for `tcfv2_corp` (Consent or Pay TCFV2)
+   - AUS for `aus`
+   - Ireland/ROW for `tcfv2`
+2. Load the CODE environment: https://m.code.dev-theguardian.com
+3. Check the correct banner is shown depending on the region
+4. Open the browser console and check if the appropriate stub is loaded or not. https://github.com/guardian/transparency-consent-docs/blob/main/docs/testing-cmp-locally.md#tools-to-test-cmp
+   - US should load **gpp but no **tcfapi
+   - UK/ROW should load **tcfapi but no **gpp
+   - AUS shouldn't have a stub.
+5. Check Google Ad Requests
+
+   - Open the Network tab in Developer tools
+   - Filter by `securepubads`
+   - Confirm if there are requests to `https://securepubads.g.doubleclick.net/gampad/ads`. Please note: for AUS and US this will be there without interacting with the banner. For GDPR (UK and ROW), this will show only after interacting with the banner.
+   - Copy the full request url to the commercial request parser (https://guardian.github.io/commercial-request-parser/)
+
+     - Check the rdp and pa values
+
+   - | Framework | RDP for Accept All | RDP for Reject All | PA for Accept All | PA for Reject All |
+     | :-------- | :----------------: | :----------------: | :---------------: | :---------------: |
+     | UK        |         na         |         na         |         t         |         f         |
```

Comment:
When signed in there are no requests at all (i.e. no advertising, as opposed to no personalised advertising).  When not signed in, it's not possible to reject all.  So I don't think we ever get a value of 'f' for 'pa' (and similarly no value of rdp for Reject All)

Reply:
Thanks